File: TreeWatch/src/main/java/com/example/demo/model/Plant.java
package com.example.demo.model;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import lombok.*;


@Entity
@Data
@NoArgsConstructor
@AllArgsConstructor
public
class Plant {
    
    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private int id;
    private String name;
    private String description;
    private int waterLevel;
    private int soilMoistureLevel;
    private int waterThreshold;
    private int soilThreshold;
    private String image;

}

File: angular-16/src/app/plant.ts
export class Plant {
    name: string;
    description: string;
    waterLevel : number;
    soilMoistureLevel : number;
    waterThreshold : number;
    soilThreshold : number;
    image : string;


    constructor(name: string, description: string, waterLevel: number, soilMostureLevel : number, waterThreshold : number, soilThreshold : number, image : string) {
        this.description = description;
        this.waterLevel = waterLevel;
        this.soilMoistureLevel = soilMostureLevel;
        this.waterThreshold = waterThreshold;
        this.soilThreshold = soilThreshold;
        this.name = name;
        this.image = image;
      }
}